dev/TEAL_opcodes.md

```diff
@@ -434,16 +501,233 @@ for notes on transaction fields available, see `txn`. If this transaction is _i_
 
 The `bnz` instruction opcode 0x40 is followed by two immediate data bytes which are a high byte first and low byte second which together form a 16 bit offset which the instruction may branch to. For a bnz instruction at `pc`, if the last element of the stack is not zero then branch to instruction at `pc + 3 + N`, else proceed to next instruction at `pc + 3`. Branch targets must be well aligned instructions. (e.g. Branching to the second byte of a 2 byte op will be rejected.) Branch offsets are currently limited to forward branches only, 0-0x7fff. A future expansion might make this a signed 16 bit integer allowing for backward branches and looping.
 
+At LogicSigVersion 2 it became allowed to branch to the end of the program exactly after the last instruction, removing the need for a last instruction or no-op as a branch target at the end. Branching beyond that may still fail the program.
```

Comment:
What does it mean "it became allowed to branch to the end of the program exactly after the last instruction"? Does this refer to the return statement? If so it needs to be changed to say it explicitly

Reply:
No--for a TEAL program with `N` bytes, a `bnz` to byte `N` (with 0-indexing) is illegal before LogicSigVersion 2 and is legal after it.

Reply:
I tried to clarify the text--let me know if it is still confusing.